src/django/api/views/v1/production_locations.py

```diff
@@ -16,19 +20,75 @@
 from api.serializers.v1.production_locations_serializer \
     import ProductionLocationsSerializer
 from api.views.v1.index_names import OpenSearchIndexNames
+from api.permissions import IsRegisteredAndConfirmed
+from api.moderation_event_actions.creation.moderation_event_creator \
+    import ModerationEventCreator
+from api.moderation_event_actions.creation.location_contribution \
+    .location_contribution import LocationContribution
+from api.moderation_event_actions.creation.dtos.create_moderation_event_dto \
+    import CreateModerationEventDTO
+from api.models.moderation_event import ModerationEvent
+from api.throttles import DataUploadThrottle
+from api.constants import (
+    APIV1CommonErrorMessages,
+    NON_FIELD_ERRORS_KEY,
+    APIV1LocationContributionErrorMessages
+)
+from api.exceptions import ServiceUnavailableException
 
 
 class ProductionLocations(ViewSet):
     swagger_schema = None
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self.opensearch_service = OpenSearchService()
-        self.opensearch_query_builder = ProductionLocationsQueryBuilder()
-        self.opensearch_query_director = OpenSearchQueryDirector(
-            self.opensearch_query_builder
+    @staticmethod
+    def init_opensearch() -> Tuple[OpenSearchService, OpenSearchQueryDirector]:
+        opensearch_service = OpenSearchService()
+        opensearch_query_builder = ProductionLocationsQueryBuilder()
+        opensearch_query_director = OpenSearchQueryDirector(
+            opensearch_query_builder
         )
 
+        return (opensearch_service, opensearch_query_director)
+
+    def get_permissions(self):
```

Comment:
I suggest to separate concerns for this method like:

```
def get_permissions(self):
    """
    Returns the list of permissions for the ViewSet action methods.
    Combines action-specific and global application-level permissions.
    """
    action_permissions = self.get_action_permissions()
    combined_permission_classes = action_permissions + self.permission_classes
    return [permission() for permission in combined_permission_classes]

def get_action_permissions(self):
    """
    Returns the list of permissions specific to the current action.
    """
    if self.action == 'create':
        return [IsRegisteredAndConfirmed]
    return []
```

The `get_permissions` method now focuses solely on combining permissions.
The `get_action_permissions` method handles action-specific permissions logic.

Reply:
Good point!
Fixed.